File: core/src/main/java/vn/loitp/core/utilities/LStringUtil.java
package vn.loitp.core.utilities;

import android.text.Html;
import android.text.Spanned;
import android.text.TextUtils;

/**
 * Created by Loitp on 5/2/2017.
 */

public class LStringUtil {
    public static String convertHTMLTextToPlainText(String htmlText) {
        Spanned spanned;
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
            spanned = Html.fromHtml(htmlText, Html.FROM_HTML_MODE_LEGACY);
        } else {
            spanned = Html.fromHtml(htmlText);
        }
        char[] chars = new char[spanned.length()];
        TextUtils.getChars(spanned, 0, spanned.length(), chars, 0);
        return new String(chars);
    }
}

File: core/src/main/java/com/core/utilities/LStringUtil.kt
package com.core.utilities

import android.text.Html
import android.text.Spanned
import android.text.TextUtils

/**
 * Created by Loitp on 5/2/2017.
 */

class LStringUtil {
    companion object {
        @Suppress("DEPRECATION")
        fun convertHTMLTextToPlainText(htmlText: String): String {
            val spanned: Spanned = if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
                Html.fromHtml(htmlText, Html.FROM_HTML_MODE_LEGACY)
            } else {
                Html.fromHtml(htmlText)
            }
            val chars = CharArray(spanned.length)
            TextUtils.getChars(spanned, 0, spanned.length, chars, 0)
            return String(chars)
        }
    }
}
